ffi/src/common.rs

```diff
@@ -218,8 +222,9 @@ pub type VerifySignatureFn = extern "system" fn(PCtxtHandle, PSecBufferDesc, c_u
 #[cfg_attr(feature = "debug_mode", instrument(skip_all))]
 #[cfg_attr(windows, rename_symbol(to = "Rust_FreeContextBuffer"))]
 #[no_mangle]
-pub unsafe extern "system" fn FreeContextBuffer(_pv_context_buffer: *mut c_void) -> SecurityStatus {
-    // FIXME: patch leak
+pub unsafe extern "system" fn FreeContextBuffer(pv_context_buffer: *mut c_void) -> SecurityStatus {
+    let _ = Box::from_raw(pv_context_buffer as *mut _);
```

Comment:
It’s unclear to me how the type is inferred at the `_` here :thinking: 
I would think it changes nothing (it gives `*mut c_void`).
What kind of data is that supposed to be? `SecBuffer`?

It’s okay to use `as *mut _` but maybe type annotation (`let _: Box<…> =`) for sanity is best?

Reply:
There does not have to be a `SecBuffer` there. The doc says that this function clears any buffer created by the security context. We do not know what type is behind that pointer. If we take FreeRDP for example, then `SecPkgInfoW /A` is deleted through `FreeContextBuffer`.

Reply:
It frees the memory the pointer points to, but doesn't call `drop()`. We can't identify what type `pv_context_buffer` contains as  `FreeContextBuffer` can be called with any buffer. 
I've checked the memory freeing with changes made in this PR with `cargo-valgrind` and there are no leaks.

Reply:
Fair enough